docs/user/guides/version-migrations.md

```diff
@@ -0,0 +1,31 @@
+# Version Migrations
+
+The [openapi-generator-cli image] version used depends on the pulpcore version.
+
+This guide list known breaking changes and recommended compatibility actions for pulpcore upgrade paths.
+
+## pulpcore `<3.70` to `>=3.70`
+
+Upgrading pulpcore from a `<3.70` to a `>=3.70` version will bump the `openapi-generator-cli` version from `4.3.1` to `7.10.0`.
+
+### Python
+
+- `Response.to_dict()|.to_json()` returns a dictionary without read-only fields.
+    - **Example**: `pulp_href` isn't present anymore.
+    - **Actions**:
+        - Use `Response.model_dump()` if Pydantic 2 is available. See [model_dump docs].
```

Comment:
Is there a way pydantic can not be there?
(It may be an option while creating the bindings but the ones this repository creates will always depend on pydantic AFAIK.)